semana7/labefy2/src/pages/CreatePlaylistPage.js

```diff
@@ -0,0 +1,134 @@
+import React from "react";
+import styled from "styled-components";
+import axios from "axios";
+
+const Section = styled.section`
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+`;
+
+const Container = styled.div`
+  padding: 1rem;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+`;
+
+const Heading2 = styled.h2`
+  padding: 1rem;
+`;
+
+const Form = styled.form`
+  display: flex;
+  align-items: center;
+  padding: 1rem;
+  width: 75%;
+`;
+
+const Input = styled.input`
+  padding: 0.5rem;
+  margin: 0.5rem 0;
+  border: 1px solid #c4d585;
+  border-radius: 5px;
+`;
+
+const Button = styled.button`
+  padding: 0.5rem 1rem;
+  border: none;
+  border-radius: 5px;
+  font-size: 1rem;
+  font-weight: 1000;
+  font-family: "Carme", sans-serif;
+  text-transform: uppercase;
+  cursor: pointer;
+  margin: 0.5rem 1rem;
+  background-color: #c1d37f;
+  color: #fff;
+  align-self: flex-end;
+
+  &:hover {
+    background-color: #b5cb67;
+  }
+`;
+
+class CreatePlaylistPage extends React.Component {
+  state = {
+    myPlaylists: [],
+    nameInputValue: "",
+  };
+
+  onChangeName = (event) => {
+    this.setState({ nameInputValue: event.target.value });
+  };
+
+  getPlaylists = async () => {
+    try {
+      const response = await axios.get(
+        "https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists",
+        {
+          headers: {
+            Authorization: "victor-cardoso-epps",
+          },
+        }
+      );
+      this.setState({ myPlaylists: response.data.result.list });
+    } catch (error) {
+      console.log(error.response);
+    }
+  };
+
+  createPlaylist = async () => {
+    const playlist = {
+      name: this.state.nameInputValue,
+    };
+    try {
+      const response = await axios.post(
+        "https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists",
+        playlist,
+        {
+          headers: {
+            Authorization: "victor-cardoso-epps",
+          },
+        }
+      );
+      alert(`${playlist.name} created`);
+      this.setState({ nameInputValue: response.data });
+      this.getPlaylists();
+    } catch (error) {
+      // if (
+      //   error.message === "There already is a playlist with a similiar name"
+      // ) {
+      //   alert("Nome da playlist já foi utilizado");
+      // }
+      console.log(`Erro: ${error.message}`);
```

Comment:
Victor, vi aqui que você tentou fazer como eu te falei! Não deu certo por causa de um detalhe que esqueci: quando usamos o axios, para capturar a mensagem de erro, precisamos colocar assim: `error.response.data.message`.

Então, se você deixar assim: 

```
if (
      error.response.data.message === "There already is a playlist with a similiar name."
   ) {
      alert("Nome da playlist já foi utilizado");
     }
```

Vai funcionar! Detalhe que a mensagem tem que ser exatamente igual a que retorna, então "There already is a playlist with a similiar name`.`" tem que ter um ponto no final, pois é assim que o back manda essa mensagem de erro.

Outro detalhe: não é legal usar template string dentro do console, porque ele não vai conseguir impimir objetos, então é melhor colocar assim: `console.log('Erro: ', error);`